src/Tizen.Peripheral/Tizen.Peripheral/Pwm.cs

```diff
@@ -0,0 +1,157 @@
+﻿/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativePwm = Interop.Peripheral.Pwm;
+
+namespace Tizen.Peripheral
+{
+    /// <summary>
+    /// The class allows applications to use the platform PWM peripheral.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class Pwm : IDisposable
+    {
+        /// <summary>
+        /// Enumeration for PWM polarity.
+        /// </summary>
+        public enum Polarity
+        {
+            /// <summary>
+            /// PWM signal start in the active high state.
+            /// </summary>
+            ActiveHigh = 0,
+
+            /// <summary>
+            /// PWM signal start in the active low state.
+            /// </summary>
+            ActiveLow
+        }
+
+        //TODO provide default values.
+        private uint _period;
+        private uint _dutyCycle;
+        private Polarity _polarity;
+        private bool _enabled;
+
+        /// <summary>
+        /// Native handle to PWM.
+        /// </summary>
+        protected IntPtr _handle;
+
+        private Pwm(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Opens the PWM pin.
+        /// </summary>
+        /// <param name="chip">The PWM chip number.</param>
+        /// <param name="pin">The PWM pin (channel) number to control.</param>
+        /// <returns>The PWM object.</returns>
+        public static Pwm Open(int chip, int pin)
+        {
+            var ret = NativePwm.Open(chip, pin, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new Pwm(handle);
+        }
+
+        /// <summary>
+        /// Closes the PWM pin.
+        /// </summary>
+        public void Close()
+        {
+            var ret = NativePwm.Close(_handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Disposes the Pwm.
+        /// </summary>
+        public void Dispose()
```

Comment:
Implement the dispose pattern
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose